geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/Transform.scala

```diff
@@ -0,0 +1,248 @@
+     * to any attribute (e.g. in AttributeKeyPlusValueIterator)
+     *
+     * @param sft simple feature type
+     * @param e expression
+     * @return
+     */
+    private def attributeExpression(sft: SimpleFeatureType, e: String): ExpressionTransform = {
+      try {
+        val expression = ECQL.toExpression(e)
+        val descriptor = expression.evaluate(sft).asInstanceOf[AttributeDescriptor]
+        val binding = if (descriptor == null) { classOf[String] } else { descriptor.getType.getBinding }
+        ExpressionTransform(e, binding, expression)
+      } catch {
+        case NonFatal(e) => throw new IllegalArgumentException(s"Unable to parse expression '$e':", e)
+      }
+    }
+
+    /**
+     * Create the feature type corresponding a transform
+     *
+     * @param sft original simple feature type
+     * @param transforms transforms
+     * @return
+     */
+    def schema(sft: SimpleFeatureType, transforms: Seq[Transform]): SimpleFeatureType = {
+      val schema = new java.util.ArrayList[AttributeDescriptor]()
+      var geom: GeometryDescriptor = null
+
+      lazy val typeBuilder = new AttributeTypeBuilder()
+
+      transforms.foreach { t =>
+        val descriptor = t match {
+          case t: PropertyTransform =>
+            val d = sft.getDescriptor(t.i)
+            val im = SimpleFeatureTypes.immutable(d)
+            if (d == sft.getGeometryDescriptor) {
+              geom = im.asInstanceOf[GeometryDescriptor]
+            }
+            im
+
+          case t: RenameTransform =>
+            sft.getDescriptor(t.i) match {
+              case d: GeometryDescriptor =>
+                val im = new ImmutableGeometryDescriptor(d.getType, new NameImpl(t.name), d.getMinOccurs,
+                  d.getMaxOccurs, d.isNillable, d.getDefaultValue, d.getUserData)
+                if (d == sft.getGeometryDescriptor) {
+                  geom = im
+                }
+                im
+              case d: AttributeDescriptor =>
+                new ImmutableAttributeDescriptor(d.getType, new NameImpl(t.name), d.getMinOccurs, d.getMaxOccurs,
+                  d.isNillable, d.getDefaultValue, d.getUserData)
+            }
+
+          case t: ExpressionTransform =>
+            typeBuilder.setBinding(t.binding)
+            if (classOf[Geometry].isAssignableFrom(t.binding)) {
+              typeBuilder.crs(CRS_EPSG_4326)
+              val typ = typeBuilder.buildGeometryType()
+              new ImmutableGeometryDescriptor(typ, new NameImpl(t.name), 0, 1, true, null, Collections.emptyMap())
+            } else {
+              val typ = typeBuilder.buildType()
+              new ImmutableAttributeDescriptor(typ, new NameImpl(t.name), 0, 1, true, null, Collections.emptyMap())
+            }
+        }
+
+        schema.add(descriptor)
+      }
+
+      var i = 0
+      while (geom == null && i < schema.size) {
+        schema.get(i) match {
+          case d: ImmutableGeometryDescriptor => geom = d
```

Comment:
So, this method makes it clear that currently Transforms cannot have multiple geometries and let the 'user' pick the default geometry. 

There's a chance we'll want to do that one day.;)  Seems like the current mappings (Transforms) could have an extra bit of side car meta-data around.  Some of that information could also explain how to set userData (such as visibilities).

Reply:
yeah there's no way to pick the default geometry - however, the default geometry doesn't mean much in this case does it? you could always style on a different geometry if you wanted.

Reply:
this seems like something that should be standardized at the geotools level.